packages/sdk-snippets/src/targets/targets.test.ts

```diff
@@ -0,0 +1,90 @@
+import type { Parameters } from '..';
+import type { ClientId, TargetId } from './targets';
+import { readdirSync, readFileSync, writeFileSync } from 'fs';
+import path from 'path';
+
+import { availableTargets, extname } from '../helpers/utils';
+import { MetricsSDKSnippet } from '..';
+
+const expectedBasePath = ['src', 'fixtures', 'parameters'];
+
+const inputFileNames = readdirSync(path.join(...expectedBasePath), 'utf-8');
+
+const fixtures: [string, Parameters][] = inputFileNames.map(inputFileName => [
+  inputFileName.replace(path.extname(inputFileName), ''),
+  // eslint-disable-next-line import/no-dynamic-require, global-require
+  require(path.resolve(...expectedBasePath, inputFileName)),
+]);
+
+/** useful for debuggin, only run a particular set of targets */
+const targetFilter: TargetId[] = [
+  // put your targetId:
+  // 'node',
+];
+
+/** useful for debuggin, only run a particular set of targets */
+const clientFilter: ClientId[] = [
+  // put your clientId here:
+  // 'express',
+];
+
+/** useful for debuggin, only run a particular set of fixtures */
+const fixtureFilter: string[] = [
+  // put the name of the fixture file you want to isolate (excluding `.js`):
+  // 'security-variables',
+];
+
+/**
+ * This is useful when you want to make a change and overwrite it for every fixture.
+ * Basically a snapshot reset.
+ *
+ * Switch to `true` in debug mode to put into effect.
+ */
+const OVERWRITE_EVERYTHING = Boolean(process.env.OVERWRITE_EVERYTHING) || false;
```

Comment:
Is there a benefit to using Jest snapshots here, instead of our own mechanism? I am okay with this, and the fixture outputs do look better than jest snapshots, but saves us from having to reinvent the wheel with this.

Reply:
This work was pulled directly over from HTTPSnippet and I kind of like it because it lets us have actual fixtures we can work with. In our HTTPSnippet fork we run those fixtures directly in the integration suite I built.

Reply:
Sounds good!